src/Workspaces/Core/Portable/Serialization/SerializableSourceText.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Serialization
+{
+    internal sealed class SerializableSourceText
+    {
+        public ITemporaryTextStorageWithName? Storage { get; }
+        public SourceText? Text { get; }
```

Comment:
doc that this is one or the other.

Reply:
➡️ Documented that both are allowed as long as they are the same